### ISSUE TITLE
Multiple Transcode Arguments per OS Type

### ISSUE THREAD

**Issue by fromage9747** [HaveAGitGat/Tdarr#531]
**Is your feature request related to a problem? Please describe.**
I have multiple Devices running different OS's as in Linux, Windows and Mac.  I am wanting to use my Handbrake custom preset but I have to choose either to use my UNIX Devices or my Windows Devices as the share path doesn't work for both platforms.  

**Describe the solution you'd like**
Ability to add transcode arguments per OS.  So for example, all Windows Devices will use one set of arguments and all Mac Devices will use another, same goes for Linux.

**Describe alternatives you've considered**
I have tried hosting my preset on an externally accessible internet share but it did not work. 

**Additional context**
![image](https://user-images.githubusercontent.com/23147783/147803411-17abd72e-ef3d-437c-8983-352c59c5d1a2.png)

**Comment by HaveAGitGat** [comment]
The pathTranslators should work for handbrake presets so put the preset in a location shared between server and node (such as in your library or transcode cache).

https://docs.tdarr.io/docs/library-setup/transcode-options/handbrake-presets

Example here:

https://github.com/HaveAGitGat/Tdarr/issues/490

**Comment by fromage9747** [comment]
@HaveAGitGat I did try this as well as it was my first thing to try and it didn't work.  I will give it another go and report back.

**Comment by fromage9747** [comment]
@HaveAGitGat 

I tried it out as per: https://github.com/HaveAGitGat/Tdarr/issues/490

 It doesn't work.  Causes all my encodes to fail.

Transcode Argument:
`--preset-import-file "F:\TdarrCache\preset\x265_10bit_Preset_CPU.json" -Z "x265_10bit_Preset_CPU"`

PathTranslator Windows:
```
{
      "server": "F:/TdarrCache",
      "node": "//mediaserver2/F - Archives/TdarrCache"
    }
```
PathTranslator Mac:
```
{
      "server": "F:/TdarrCache",
      "node": "/Volumes/TdarrCache"
    }
```
The Tdarr Server is sitting on a Widows Device, so the path translation works for Windows, but not anything else that is not Windows.

**Comment by HaveAGitGat** [comment]
Try entering the preset path with forward slashes as in the example. 

You’ve used backward slashes in the path but the translator uses forward slashes so the translation isn’t working. Hopefully that’s it.

**Comment by fromage9747** [comment]
@HaveAGitGat Thank you for getting back to me.  I tried it with forwarding slashes and the result is the same. I continued to try different permutations of the path but none of them changed the result. 

The Mac Device cannot read the preset file due to not being able to access it.  So I can either use only Windows Devices as server and node with my handbrake preset or only Unix Devices as server and node.

**Comment by fromage9747** [comment]
Today I setup my Linux Tdarr server and tried to connect my Mac to it. Same issue with the handbrake preset file. Pretty much my feature request is valid. If there could be additional argument fields per OS, this would solve this problem.

**Comment by fromage9747** [comment]
@HaveAGitGat From what I can determine is that the path translators work perfectly fine for what Tdarr has control over as in the paths of the transcode cache and the source files to be transcoded.  The path translator cannot handle paths outlined in arguments from plugins and why would it?  There can be any number of arguments placed in that field, how is Tdarr to identify which argument is a path and which is not? 

Just to summarise. 
1. Mac can connect to Windows Tdarr server using Path Translators for setting the paths under Source, Transcode Cache and Output path:
![image](https://user-images.githubusercontent.com/23147783/149064834-64781625-ae03-4d69-b9a7-157d5d085c9e.png)
2. Mac can connect to Linux Tdarr server using Path Translators for setting the paths under Source, Transcode Cache and Output path:
![image](https://user-images.githubusercontent.com/23147783/149064898-5d80476e-1fed-40b3-b327-975633c49abe.png)

But when you specify a path in the Transcode options Plugin Argument, path translators do not work as I am pretty sure that they have not been coded to do so:

![image](https://user-images.githubusercontent.com/23147783/149065029-105d3f1c-7a3a-4901-9be4-78bc4489dc54.png)

My feature request is to have the ability to add multiple Argument fields and be able to map them to each node.  This would be the easiest way, as to implement some type of Device OS type functionality would require more time to do.  But being able to add 3 arguments and then under each node have the option to map what argument in the plugin would be the easiest and quickest way to implement.

**Comment by supersnellehenk** [comment]
So, I can confirm that the translators aren't working as expected here. They have worked in previous versions, I'm sure of that. 

Creating individual plugin entries based upon their OSes is kinda tricky, although I'm not sure if the node executes the plugin or the server does it. Also incorporating this into a UI is pretty ugly in most ways, that would require a pretty major redesign of the transcode stack page to fit nicely.

I think the translators not working is a bug which should be fixed instead of a new feature which *technically* already has a solution, but it's not working at the moment.

**Comment by fromage9747** [comment]
Alright, at least I haven't lost my mind with this. My implementation of it would be per node. Use default plugin config from parent of specify node specific plugin config. I could draw a mock-up of it when at my desk in the morning. Anyway, I would be happy if the bug was fixed. Cheers!

**Comment by fromage9747** [comment]
Dam, I didn't think of that.  Thank you! I shall do some fiddling

**Comment by fromage9747** [comment]
Finally got round to implementing this today and it's working like a charm. Cheers

**Comment by loghan-cruel** [comment]
I use windows and I have the same problem, how do I edit line 205?

**Comment by loghan-cruel** [comment]
I edited as follows     https://prnt.sc/QvBG0-dP__QM

**Comment by HaveAGitGat** [comment]
Went through this with someone on Discord and fixed the path translation issue above for 2.00.20.

Regarding dynamic platform arguments, at the moment those should be handled using the `hardware encoding type` Node option on the node options panel and conditional flows in the plugin such as `if (os.platform() === 'win32') {` or `if(otherArguments.nodeHardwareType === 'nvenc'){`

Futher node/worker tagging will be handled in:
https://github.com/HaveAGitGat/Tdarr/issues/748
